controllers/v1/usersController.js

```diff
@@ -109,6 +127,98 @@ module.exports = class Users {
     });
   }
 
+  /**
+   * @api {post} /survey/v1/users/solutions/:programId?page=:page&limit=:limit&search=:searchText
+   * @apiVersion 1.0.0
+   * @apiName User solutions
+   * @apiGroup Users
+   * @apiHeader {String} X-auth-token Authenticity token
+   * @apiSampleRequest /survey/v1/users/solutions/5ff438b04698083dbfab7284?page=1&limit=10
+   * @apiParamExample {json} Request-Body:
+   * {
+        "state": "665d8df5c6892808846230e7",
+        "district": "668240135fb8bc3e93ceae39",
+        "block": "6682771aa845ef3e891db070",
+        "cluster": "668242835fb8bc3e93ceae44",
+        "role": "mentee,program_creator,block_education_officer,principle,district_education_officer"
+    }
+   * @apiUse successBody
+   * @apiUse errorBody
+   * @apiParamExample {json} Response:
+   * {
+    "message": "Program solutions fetched successfully",
+    "status": 200,
+    "result": {
+        "programName": "Testing by dev team ap",
+        "programId": "669a9836c5da53e877b8969e",
+        "description": "testing by dev team",
+        "rootOrganisations": "",
+        "data": [
+            {
+                "_id": "66c2eb3e39f69eb48f2980f7",
+                "externalId": "Test-survey2-1724050238923",
+                "name": "Test survey 1",
+                "description": "Test survey 1",
+                "language": [
+                    "English"
+                ],
+                "type": "survey",
+                "endDate": "2026-12-25T00:00:00.000Z",
+                "link": "abad9308b93222bb9d0ccc8e79efdbe8",
+                "entityType": ""
+            },
+            {
+                "_id": "66d055bbdf92822366029c1c",
+                "externalId": "606d92fa-42d8-11ec-ac61-29082024-15-57-OBSERVATION-TEMPLATE-1724929467852",
+                "name": "dev_testing",
+                "description": "dev testing",
+                "language": [
+                    "English"
+                ],
+                "entityType": "school",
+                "type": "observation",
+                "endDate": "2025-08-29T11:04:27.852Z",
+                "link": "464f9f5d68282096a190b583748cf8ec"
+             }
+        ],
+        "count": 2
+      }
+    }
+   **/
+
+  /**
+   * User targeted solutions.
+   * @method
+   * @name solutions
+   * @param  {req}  - requested data.
+   * @returns {json} List of targeted solutions.
+   */
+
+  solutions(req) {
```

Comment:
@praveenKDass have we added configurations for these new apis? If not please add

Reply:
added @VISHNUDAS-tunerlabs